crates/ruff_linter/src/rules/ruff/rules/dummy_variable_accessed.rs

```diff
@@ -0,0 +1,246 @@
+        })
+        .collect();
+
+    // Try at most one auto-fix for first diagnostic instance
+    if let Some(fix_kind) = possible_fix_kind {
+        // Get the possible fix based on the scope
+        if let Some(fix) = get_possible_fix(name, fix_kind, scope) {
+            // Fix is available
+            if let Some(diagnostic) = diagnostics.first_mut() {
+                // Try to set the fix
+                diagnostic.try_set_fix(|| {
+                    let (edit, rest) =
+                        Renamer::rename(name, &fix, scope, checker.semantic(), checker.stylist())?;
+                    let applicability = Applicability::Safe;
+                    Ok(Fix::applicable_edits(edit, rest, applicability))
+                });
+            }
+        }
+    }
+
+    Some(diagnostics)
+}
+
+/// Enumeration of various ways in which a binding can shadow other variables
+#[derive(Debug, PartialEq, Eq, Copy, Clone)]
+enum ShadowedKind {
+    /// The variable shadows a global, nonlocal or local symbol
+    Some,
+    /// The variable shadows a builtin symbol
+    BuiltIn,
+    /// The variable shadows a keyword
+    Keyword,
+    /// The variable does not shadow any other symbols
+    None,
+}
+
+/// Suggests a potential alternative name to resolve a shadowing conflict.
+fn get_possible_fix(name: &str, kind: ShadowedKind, scope: &Scope) -> Option<String> {
+    // Remove leading underscores for processing
+    let trimmed_name = name.trim_start_matches('_');
+
+    // Construct the potential fix name based on ShadowedKind
+    let fix_name = match kind {
+        ShadowedKind::Some | ShadowedKind::BuiltIn | ShadowedKind::Keyword => {
+            format!("{trimmed_name}_") // Append an underscore
+        }
+        ShadowedKind::None => trimmed_name.to_string(),
+    };
+
+    // Ensure the fix name is not already taken in the scope
+    if scope.has(&fix_name) {
+        return None;
+    }
+
+    // Check if the fix name is a valid identifier
+    is_identifier(&fix_name).then_some(fix_name)
+}
+
+/// Determines the kind of shadowing or conflict for a given variable name.
+fn get_possible_fix_kind(name: &str, scope: &Scope, checker: &Checker) -> Option<ShadowedKind> {
+    // If the name starts with an underscore, we don't consider it
+    if !name.starts_with('_') {
+        return None;
+    }
+
+    // Trim the leading underscores for further checks
+    let trimmed_name = name.trim_start_matches('_');
+
+    // Check the kind in order of precedence
+    if is_keyword(trimmed_name) {
+        return Some(ShadowedKind::Keyword);
+    }
+
+    if is_python_builtin(
+        trimmed_name,
+        checker.settings.target_version.minor(),
+        checker.source_type.is_ipynb(),
+    ) {
+        return Some(ShadowedKind::BuiltIn);
+    }
+
+    if scope.has(trimmed_name) {
+        return Some(ShadowedKind::Some);
+    }
+
+    // Default to no shadowing
+    Some(ShadowedKind::None)
+}
+
+fn is_keyword(name: &str) -> bool {
```

Comment:
You can use this existing function in our `ruff_python_stdlib` crate rather than implementing your own:

https://github.com/astral-sh/ruff/blob/be07424e808b667c57e366c645c8aa349257cb7d/crates/ruff_python_stdlib/src/keyword.rs#L2

Reply:
Thanks for checking. Here I think that's fine